### PR TITLE
docs(monitoring): update metrics.md - add Service Monitors for dex, notifications and redis

### PR DESCRIPTION
**Pull request by dmpe**
This has been tested with almost latest `argocd 2.9.2` and latest prometheus operator.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

File: docs/operator-manual/metrics.md [hunk 1]
```diff
@@ -148,6 +148,55 @@ spec:
...
+spec:
+  endpoints:
+  - port: http-exporter-port
+    path: /metrics
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: argocd-redis-ha-haproxy
+```
+
+For notification controller, you additionally need to add following:
```
Comment (dmpe) [inline docs/operator-manual/metrics.md:181]:
```suggestion
For notification controller, you need to additionally add following: 
```

File: docs/operator-manual/metrics.md [hunk 2]
```diff
@@ -148,6 +148,55 @@ spec:
...
+```
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: argocd-redis-haproxy-metrics
+  namespace: argocd
+spec:
+  endpoints:
+  - port: http-exporter-port
+    path: /metrics
```
Comment (dmpe) [inline docs/operator-manual/metrics.md:175]:
```suggestion
```

File: docs/operator-manual/metrics.md [hunk 3]
```diff
@@ -148,6 +148,54 @@ spec:
   - port: metrics
 ```
 
+```
```
Comment (chengfang) [inline docs/operator-manual/metrics.md:151]: can you use ```yaml for pretty formatting?

**Comment by dmpe** [comment]
Thank for review, will do within the next 24 hours ...

**Comment by dmpe** [comment]
> Thank for review, will do within the next 24 hours ...

I have add some smaller fixes. Can you take a look again pls :) Many thanks.

**Review by chengfang — APPROVED** [review]

**Comment by dmpe** [comment]
Commits squashed, rebased and pushed again.

**Review by blakepettersson — APPROVED** [review]

**Comment by dmpe** [comment]
Can someone merge this ?

**Comment by blakepettersson** [comment]
Sorry missed the actual merge, LGTM! 🎉

**Pull request merged**